content/structure.md

```diff
@@ -0,0 +1,390 @@
+
+<h2 id="javascript-structure">JavaScript file structure</h2>
+
+To harness the module system and ensure that our code only runs when we specify that it should, all code for libraries, components and APIs should be placed in the `imports/` directory. This will mean that the Meteor build system will only bundle and include that code if it is `import`-ed from another file.
+
+Meteor will eagerly load any files outside of `imports/` in the application, but to be explicit, it's best to create a `main/client` and `main/server` folder to define explicit entry points for the running code of your application. 
+
+In the Todos example app, the `main/client/routes.js` indirectly file imports *all* other code that is required on the client, and forms the single entry point for the client application:
+
+```js
+import { FlowRouter } from 'meteor/kadira:flow-router';
+import { BlazeLayout } from 'meteor/kadira:blaze-layout';
+import { AccountsTemplates } from 'meteor/useraccounts:core';
+
+// This file needs to load first, so that useraccounts:core doesn't complain when we define
+// a forgotPwd route
+import '../useraccounts-configuration.js';
+
+// Import to load these templates
+import '../../imports/client/layouts/app-body.js';
+import '../../imports/client/pages/root-redirector.js';
+import '../../imports/client/pages/lists-show-page.js';
+import '../../imports/client/pages/app-not-found.js';
+
+// Import to override accounts templates
+import '../../imports/client/accounts/accounts-templates.js';
+
+// Below here are the route definitions
+```
+
+On the server, there can often be a few more logical entry points to control the behavior of the application server. In the Todos app, we define four:
+
+1. `main/server/api.js' - this defines all the collections, publications and methods that the application provides as an API to the client.
+2. `main/server/fixtures.js' - this defines a starting set of data to be loaded if the app is loaded with an empty db.
+3. `main/server/reset-password-email.js` - this file configures the Accounts package to define the UI of the reset password email.
+4. `main/server/security.js` - setups some rate limiting and other important security settings.
+
+Other logical server-side entry points include setting up job workers and server-side routes.
+
+<h3 id="structuring-imports"</h3>
+
+Once you've placed all files in the `imports/` directory, it makes sense to start thinking about how best to organize it. A good starting point is splitting data / business logic related code from rendering code. We suggest an `imports/api` and `imports/client` as a logical split. 
+
+XXX: I don't think `imports/client` is a good name given server-side rendering. `imports/view[s]`?
```

Comment:
Ok I turned on `METEOR_PROFILE` and the rebuild time when editing a component file is ~ 2.5s on my machine in both cases. If anything it was a little faster after I renamed the directory to `import/views`. So let's say it doesn't matter?

So what do we think the best directory name is?